### DISCUSSION TITLE
LunaPark Visual scripting and php framework connection

### DISCUSSION THREAD

**Discussion by ivnlabs** [lunapark/lunapark#1] [General]
Hi, is it possible to connect the Luna-Park visual scripting to a PHP development running with the Laravel Framework? I was imagining working with the nodes and wires connecting the Laravel functions, if this is possible I want to be the first to test it! I am new to the lunaPark app but so far what I have seen I like it! great job

thanks in advance, 
Daniel

**Comment by hugoattal** [comment]
Hey Daniel!

That's a very interesting usecase 🤔. You could actually integrate the Luna Park editor into a Laravel Framework if I ship a standalone Luna Park js script (but that's not the case yet, so for now, you need a node package manager to install it through NPM: https://www.npmjs.com/package/luna-park)

You would then need to add it to your HTML using a `<script>` tag. Then, you could use a custom HTML element this this:

```html
<lp-logic-editor editor-id="myCustomEditor" playground />
```
(check the instructions here: https://luna-park.app/documentation/editor-integration/basic-integration)

However, you would need a bit of JS to save and load the logic made by users. And the compiled version of the logic would be in JS. If you intend to use the compiled logic on the client side, there's no problem. But if you want to trigger the logic on the PHP server, that would require either :
- A node.js installation to run the Luna Park JS sandbox
- A transpilation of the compiled JS into PHP, but you would need to build your own sandbox

Anyhow, I'm super interesting to discuss your usecase and see what you would need 🙂 !
You can hit me up on Discord or LinkedIn (https://www.linkedin.com/in/hugoattal)
